docs/src/references/agent/operation-workflow.md

```diff
@@ -437,6 +492,37 @@ on_timeout = "timeout_restart"
 on_success = "successful_restart"
 ```
 
+#### Awaiting Sub-Command Completion
+
+When a sub `command` is executed, the operation workflow moves to a waiting state which `action` must be `await-command-completion`.
+This waiting state specifies, with `on_success`, `on_error` and `on_timeout` handlers,
+the next step of the calling workflow on success, failure or timeout of the sub-command.
+
+The `await-command-completion` action also tells how to inject back the outcome of the sub-command into the invoking command state.
+- This is done by attaching values to `output` subfields, as in `output.x.y.z = "${.payload.x.y.z}"`
+- Many independent subfields can be added. They will all be injected into the current payload of the command state, letting untouched the others.
+- The invoking command payload cannot be overridden as a whole. Subfields can be added and updated but not removed.
+- Specific path expressions can be used to extract specific excerpts from the sub-command final state.
+- `${.}` denotes the whole sub-command final message including the `topic` and the `payload`.
+- `${.topic}` is the sub-command request topic
+- `${.topic.target}` is the sub command target identity
+- `${.topic.operation}` is the sub command operation
+- `${.topic.cmd_id}` is the sub command request unique identifier
+- `${.payload}` is the whole sub command json payload
+- `${.payload.status}` is the sub command final status
+- `${.payload.x.y.z}` is the json value extracted from the payload following the given `x.y.z` path if any.
+- If given `${.some.unknown.path}`, the subfield is assigned the unchanged expression.
+
+```toml
+[waiting_for_config_update]
+action = "await-command-completion"
+output.some.new.command.field = "${.payload.of.the.sub.command}"
```

Comment:
I found `output` to be a bit confusing. It depends on the perspective I guess. Since it is the `input` to the next state derived from the output/payload of the subcommand, I'd have just used `input` here as well. Or to be explcit, `on_success.input` with the value having references to the `payload`.

Reply:
We have to discuss this naming.

I think we have to take the (non yet implemented) view where the input and output of a command are given in a single state (the awaiting state being implied). As in:

```toml
operation = "sub-operation"
input.x = "${.excerpt.of.the.invoking.command.to.use.as.init.state}"
input.y = "${.another.excerpt.for.the.init.state}"
output.a = "${.exceprt.of.the.sub.command.final.state.to.inject.into.the.command.state}"
output.b = "${.another.excerpt.of.the.sub.command.final.state}"
```

Reply:
Just wondering if, instead of defining the `output` at the `sub-operation` level, we can define the same as `input` of the next `on_success` state.

Reply:
> Just wondering if, instead of defining the `output` at the `sub-operation` level, we can define the same as `input` of the next `on_success` state.

Yeah, I see what you mean. However, there are perhaps several states with a transition to the `on_success` state and this state payload might have to be computed differently in each case.

Reply:
Noticing that I always refer to *init* and *final* state of the sub-command, I see as an alternative the following:

```toml
operation = "sub-operation"
init.x = "${.excerpt.of.the.invoking.command.to.use.as.init.state}"
init.y = "${.another.excerpt.for.the.init.state}"
final.a = "${.exceprt.of.the.sub.command.final.state.to.inject.into.the.command.state}"
final.b = "${.another.excerpt.of.the.sub.command.final.state}"
```

Reply:
Added as a follow-up task.